modules/system/applications.nix

```
@@ -37,11 +38,40 @@ in
         rm ~/Applications/'Nix Apps'
       fi
 
-      if [ ! -e '/Applications/Nix Apps' ] \
-         || ourLink '/Applications/Nix Apps'; then
-        ln -sfn ${cfg.build.applications}/Applications '/Applications/Nix Apps'
+      ourFolder () {
+        local marker
+        marker=$(xattr -p org.nix-darwin "$1" 2>/dev/null)
+        [ "$marker" = 'generated' ]
+      }
+
+      targetFolder='/Applications/Nix Apps'
+
+      # Clean up old style link to nix store
+      if [ -e "$targetFolder" ] && ourLink "$targetFolder"; then
+        rm "$targetFolder"
+      fi
+
+      if [ -e "$targetFolder" ] && ! ourFolder "$targetFolder"; then
+        echo "warning: $targetFolder is not owned by nix-darwin, skipping App linking..." >&2
       else
-        echo "warning: /Applications/Nix Apps is not owned by nix-darwin, skipping App linking..." >&2
+        # create and mark folder
+        mkdir -p "$targetFolder"
+        xattr -w org.nix-darwin generated "$targetFolder"
+
+        rsyncFlags=(
+          --archive
+          --checksum
+          --chmod=-w
+          --copy-unsafe-links
+          --delete
+          --no-group
+          --no-owner
+        )
+
+        # sync applications
+        ${lib.getBin pkgs.rsync}/bin/rsync \
```

Could `lib.getExe` be used here?
```suggestion
        ${lib.getExe pkgs.rsync} \
```

---

Good point, I just didn't think of it. ✅ Changed